src/components/form/range/range_highlight.tsx

```diff
@@ -39,16 +42,16 @@ export const EuiRangeHighlight: FunctionComponent<EuiRangeHighlightProps> = ({
   compressed,
   background,
   onClick,
+  levels,
 }) => {
   // Calculate the width the range based on value
   // const rangeWidth = (value - min) / (max - min);
   const leftPosition = (lowerValue - min) / (max - min);
   const rangeWidth = (upperValue - lowerValue) / (max - min);
 
-  const rangeWidthStyle = {
-    background,
-    marginLeft: `${leftPosition * 100}%`,
-    width: `${rangeWidth * 100}%`,
+  const [trackWidth, setTrackWidth] = useState(0);
+  const handleRef = (node: HTMLDivElement | null) => {
+    setTrackWidth(node?.clientWidth ?? 0);
   };
```

Comment:
[optional microperf] useCallback
```suggestion
  const handleRef = useCallback((node: HTMLDivElement | null) => {
    setTrackWidth(node?.clientWidth ?? 0);
  }, []);
```

Reply:
When I `useCallback `, in the following [example](https://eui.elastic.co/pr_6265/#/forms/range-sliders#inputs-with-range-in-a-dropdown) the highlight doesn't get the right width:

<img width="421" alt="Screenshot 2022-10-25 at 15 07 01" src="https://user-images.githubusercontent.com/2750668/197797909-d2ca53f7-ab15-4329-87df-68bde3ae7649.png">

Reply:
ahh super interesting. I was wondering how the width updates were responding to changes 😅 it's not technically the most correct use of React but I'm hoping to avoid detecting clientWidth via percentages in any case - will investigate!